internal/cmd/commands/server/server.go

```diff
@@ -124,6 +126,7 @@ func (c *Command) AutocompleteFlags() complete.Flags {
 }
 
 func (c *Command) Run(args []string) int {
+	defer c.CancelFn()
```

Comment:
Note that this might be racy. If it's important that we ensure that any cleanup happens before the process exits, then we need to wait for the cleanup to be verified finished before exiting. Based on the comment below it sounds like this is best-effort anyways, but want to raise the issue just in case.